__tests__/app.test.js

```diff
@@ -347,3 +422,37 @@ describe('GET api users', () => {
       });
   });
 });
+
+describe('GET api articles queries', () => {
+  describe(('Topic query'), () => {
+    test('GET:200 sends an array of users that have the queried topic', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch')
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(Array.isArray(articles)).toBe(true);
+        expect(articles).toHaveLength(12);
+        articles.forEach((article) => {
+          expect(article.topic).toBe('mitch');
+        });
+      });
+    });
+    test('GET:404 sends an appropriate status and error message when given a non existing topic', () => {
+      return request(app)
+        .get('/api/articles?topic=banana')
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe('article does not exist');
+        });
+    });
```

Comment:
We need one more 200 test - for a topic which exists but has no articles (i.e. returns empty array)